gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -269,23 +270,40 @@ public boolean exists(URI uri) {
   }
 
   @Override
-  public Spec getSpec(URI uri) throws SpecNotFoundException {
+  public Spec getSpecs(URI uri) throws SpecNotFoundException {
     try {
       return specStore.getSpec(uri);
     } catch (IOException e) {
       throw new RuntimeException("Cannot retrieve Spec from Spec store for URI: " + uri, e);
     }
   }
 
+  @Override
+  public Collection<Spec> getSpecs(SpecSearchObject specSearchObject) {
+    try {
+      return specStore.getSpecs(specSearchObject);
+    } catch (IOException e) {
+      throw new RuntimeException("Cannot retrieve Spec from Spec store for URI: " + specSearchObject, e);
+    }
+  }
+
+  public Collection<Spec> getAllSpecs() {
+    try {
+      return specStore.getSpecs();
```

Comment:
An option to consider: GobblinServiceScheduler already caches all the specs in memory. So we should be able to avoid going to the SpecStore to retrieve all the specs each time getAllSpecs() is invoked.

Reply:
Yes, good consideration. Also need to pay attention if slave nodes also have all the specs in their memory.